### PR TITLE
Added MBConv Block

### PR DESCRIPTION
**Pull request by sayannath**
I have added the MBConv Block. 

Output after each layer
```
(None, 112, 112, 64)
(None, 56, 56, 96)
(None, 28, 28, 192)
```

Note: Print Statements are written for dev purposes.

- [X] Stem Block
- [X] MBConv Block
- [ ] Relative Attention Block
- [ ] FFN Block
- [ ] Output Block

### CODE REVIEW

File: coatnet/models/blocks.py [hunk 1]
```diff
@@ -22,3 +27,155 @@ def conv_3x3(input_layer, num_filters: int = 64, downsample: bool = False):
...
+        self,
+        input_channel: int,
+        output_channel: int,
+        se_ratio: float,
+        strides: int,
+        conv_shortcut: bool,
+        expansion: int,
+        dropout_rate: float,
+        **kwargs
+    ):
```
Comment (sayakpaul) [inline coatnet/models/blocks.py]: Missing docstrings.

File: coatnet/models/blocks.py [hunk 2]
```diff
@@ -22,3 +27,155 @@ def conv_3x3(input_layer, num_filters: int = 64, downsample: bool = False):
...
+        super(MbConv_Block, self).__init__(**kwargs)
+
+        self.bn = keras.layers.BatchNormalization()
+        self.act = tfa.layers.GELU()
+
+        if conv_shortcut:
+            if strides > 1:
+                self.conv_shortcut_pool = keras.layers.AveragePooling2D(
+                    pool_size=(strides, strides), padding="same"
+                )
```
Comment (sayakpaul) [inline coatnet/models/blocks.py]: Usually the `layers` module is imported and is used throughout. 

```py
from tensorflow import keras
from tensorflow.keras import layers
```

File: coatnet/models/blocks.py [hunk 3]
```diff
@@ -22,3 +27,155 @@ def conv_3x3(input_layer, num_filters: int = 64, downsample: bool = False):
...
+        x = self.bn_1(x)
+        x = self.act_1(x)
+        x = self.dconv_1[0](x)
+        x = self.bn_2(x)
+        x = self.act_2(x)
+        x = self.se_module(x)
+        x = self.conv_2(x)
+        x = self.drop_1(x)
+        x = keras.layers.Add()([x, shortcut])
+        return x
```
Comment (sayakpaul) [inline coatnet/models/blocks.py]: Can't we make the class a function instead? That way the end model becomes fully expandable.

If we wrap a block like this inside a `layers.Layer` then we also lose access to its internals. So, prefer making this block with the Functional API. Here's an example:

https://github.com/keras-team/keras/blob/f95c9050ae577d97f2a7cae84a3096bbc9a1d869/keras/applications/convnext.py#L238-#L297

File: coatnet/models/blocks.py [hunk 4]
```diff
@@ -22,3 +27,155 @@ def conv_3x3(input_layer, num_filters: int = 64, downsample: bool = False):
...
+
+
+class SE_Module(keras.layers.Layer):
```
Comment (sayakpaul) [inline coatnet/models/blocks.py]: Same for this + docstrings missing.

File: coatnet/models/blocks.py [hunk 5]
```diff
@@ -22,3 +27,155 @@ def conv_3x3(input_layer, num_filters: int = 64, downsample: bool = False):
...
+        self,
+        num_filters: int,
+        ratio: int = 16,
+        divisor: int = 8,
+        activation="gelu",
+        channel_format="channels_last",
+        use_bias: bool = True,
+        **kwargs
+    ):
+        super(SE_Module, self).__init__(**kwargs)
```
Comment (sayakpaul) [inline coatnet/models/blocks.py]: Nit: In Python 3, we don't need this. `super().__init__(**kwargs)` works.

**Review by sayakpaul — CHANGES_REQUESTED** [review]

File: coatnet/models/blocks.py (hunk 3 above)
Comment (sayannath) [reply coatnet/models/blocks.py]: Cool! Will try to implement it using Functional API.

Comment (sayannath) [reply coatnet/models/blocks.py]: @sayakpaul If I am not wrong this is [MBConv](https://github.com/keras-team/keras-applications/blob/06fbeb0f16e1304f239b2296578d1c50b15a983a/keras_applications/efficientnet.py#L119) Block right?

Comment (sayakpaul) [reply coatnet/models/blocks.py]: Yes, I think so. We can reuse it with due credits. @gustheman WDYT?

Comment (sayannath) [reply coatnet/models/blocks.py]: If @gustheman is okay with it. I will modify the code and make the changes.

File: coatnet/models/blocks.py [hunk 6]
```diff
@@ -22,3 +26,90 @@ def conv_3x3(input_layer, num_filters: int = 64, downsample: bool = False):
...
+    output_channel: int,
+    strides: int = 1,
+    expansion_rate: int = 4,
+    se_ratio: float = 0.25,
+    dropout_rate: float = 0.1,
+):
+    """
+    Mobile Inverted Block - Stage 2
+
+    Refer: https://github.com/keras-team/keras-applications/blob/master/keras_applications/efficientnet.py
```
Comment (sayakpaul) [inline coatnet/models/blocks.py]: Prefer providing the exact code block link.

**Review by sayakpaul — APPROVED** [review]
LGTM!

File: coatnet/models/blocks.py (hunk 6 above)
Comment (sayannath) [reply coatnet/models/blocks.py]: Cool

**Pull request merged**